README.md

```diff
@@ -9,16 +9,54 @@ Install Node + NPM
 
 ## Setup
 
-`npm install`
+- Install Dependencies
+
+  `npm install`
+
+- Initialize Submodule (https://github.com/hypertrace/hypertrace-core-ui/)
+
+  `git submodule update --init`
 
 ## Development server
 
-Run `npm start` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
+Run `npm run start` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
 
 ## Running unit tests
 
 Run `npm run test` to execute the unit tests via Jest
 
-## Commit
+## Technologies
+
+1. Angular
+2. Typescript
+3. RxJS
+4. [D3](https://d3js.org/)
+5. [Spectator](https://github.com/ngneat/spectator) (_Unit Testing_)
+
+## UI Architecture
+
+| <img src="https://hypertrace-docs.s3.amazonaws.com/ui-architecture.png" width="400" height="400"/> |
+| :------------------------------------------------------------------------------------------------: |
+|                                    _Hypertrace UI Architecture_                                    |
+
+## Submodules
+
+Pull all changes in the repository including changes in the submodules
+
+    git pull --recurse-submodules
+
+Pull all changes for the submodules
+
+    git submodule update --remote
+
+## Building Image locally
+
+Hypertrace UI uses gradlew to compile/install/distribute. Gradle wrapper is already part of the source code. To build Hypertrace UI image, run:
+
+```
+./gradlew clean build dockerBuildImages
```

Comment:
`./gradlew dockerBuildImages` is sufficient. There's no need to specify intermediate targets and clean should never be required to build, gradle's smart enough to rebuild/reuse as appropriate.